Managers/LuaMan.cpp

```diff
@@ -1,2829 +1,474 @@
+			return -1;
+		}
+		// Do not open paths that aren't written correctly
+		if (!System::PathExistsCaseSensitive(std::filesystem::path(fileName).lexically_normal().generic_string())) {
+			return -1;
+		}
+		// Allow to edit files only inside .rte folders
+		if (fullPath.find(System::GetModulePackageExtension()) == std::string::npos) {
+			return -1;
+		}
 
-	// Allow to edit files only inside .rte folders
-	if (fullPath.find(rteString) == string::npos)
+		if (FILE * file = fopen(fullPath.c_str(), accessMode.c_str())) {
+			m_OpenedFiles.at(fileIndex) = file;
+			return fileIndex;
+		}
+		g_ConsoleMan.PrintString("Error: Failed to open file " + fullPath);
 		return -1;
-
-	// Open file and save handle
-	FILE * f = fopen(fullPath.c_str(), mode.c_str());
-	if (f)
-	{
-		m_Files[fl] = f;
-		return fl;
 	}
 
-#ifdef _DEBUG
-	g_ConsoleMan.PrintString("Error: Can't open file. " + fullPath);
-#endif
-	return -1;
-}
-
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void LuaMan::FileClose(int file)
-{
-	if (file > -1 && file < MAX_OPEN_FILES && m_Files[file])
-	{
-		fclose(m_Files[file]);
-		m_Files[file] = 0;
+	void LuaMan::FileClose(int fileIndex) {
+		if (fileIndex > -1 && fileIndex < c_MaxOpenFiles && m_OpenedFiles.at(fileIndex)) {
+			fclose(m_OpenedFiles.at(fileIndex));
+			m_OpenedFiles.at(fileIndex) = nullptr;
+		}
 	}
-}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void LuaMan::FileCloseAll()
-{
-	for (int file = 0 ; file < MAX_OPEN_FILES ; ++file)
-		FileClose(file);
-}
+	void LuaMan::FileCloseAll() {
+		for (int file = 0; file < c_MaxOpenFiles; ++file) {
+			FileClose(file);
+		}
+	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-std::string LuaMan::FileReadLine(int file)
-{
-	if (file > -1 && file < MAX_OPEN_FILES && m_Files[file])
-	{
-		char buf[4096];
-		fgets(buf, 4095, m_Files[file]);
-		return (std::string(buf));
+	std::string LuaMan::FileReadLine(int fileIndex) {
+		if (fileIndex > -1 && fileIndex < c_MaxOpenFiles && m_OpenedFiles.at(fileIndex)) {
+			char buf[4096];
+			fgets(buf, 4095, m_OpenedFiles.at(fileIndex));
+			return (std::string(buf));
+		}
+		g_ConsoleMan.PrintString("Error: Tried to read a closed file, or read past EOF.");
+		return "";
 	}
-	g_ConsoleMan.PrintString("Error: Tried to read a closed file, or read past EOF.");
-	return "";
-}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void LuaMan::FileWriteLine(int file, std::string line)
-{
-	if (file > -1 && file < MAX_OPEN_FILES && m_Files[file])
-		fputs(line.c_str(), m_Files[file]);
-	else
-		g_ConsoleMan.PrintString("Error: Tried to write to a closed file.");
-}
+	void LuaMan::FileWriteLine(int fileIndex, const std::string &line) {
```

Comment:
Presumably this can't be a string_view?

Reply:
String_view doesn't have `c_str()` for `fputs` so that just means constructing a string out of the string_view to then construct a c-string.
\<insert xzibit meme here>